notify/msteamsv2/msteamsv2.go

```diff
@@ -0,0 +1,193 @@
+	retrier      *notify.Retrier
+	webhookURL   *config.SecretURL
+	postJSONFunc func(ctx context.Context, client *http.Client, url string, body io.Reader) (*http.Response, error)
+}
+
+// https://learn.microsoft.com/en-us/connectors/teams/?tabs=text1#adaptivecarditemschema
+type Content struct {
+	Schema  string `json:"$schema"`
+	Type    string `json:"type"`
+	Version string `json:"version"`
+	Body    []Body `json:"body"`
+}
+
+type Body struct {
+	Type   string `json:"type"`
+	Text   string `json:"text"`
+	Weight string `json:"weigth,omitempty"`
+	Size   string `json:"size,omitempty"`
+	Wrap   bool   `json:"wrap,omitempty"`
+	Style  string `json:"style,omitempty"`
+	Color  string `json:"color,omitempty"`
+}
+
+type Attachment struct {
+	ContentType string  `json:"contentType"`
+	ContentURL  *string `json:"contentUrl"` // Use a pointer to handle null values
+	Content     Content `json:"content"`
+}
+
+type teamsMessage struct {
+	Type        string       `json:"type"`
+	Attachments []Attachment `json:"attachments"`
+}
+
+// New returns a new notifier that uses the Microsoft Teams Power Platform connector.
+func New(c *config.MSTeamsV2Config, t *template.Template, l log.Logger, httpOpts ...commoncfg.HTTPClientOption) (*Notifier, error) {
+	client, err := commoncfg.NewClientFromConfig(*c.HTTPConfig, "msteamsv2", httpOpts...)
+	if err != nil {
+		return nil, err
+	}
+
+	n := &Notifier{
+		conf:         c,
+		tmpl:         t,
+		logger:       l,
+		client:       client,
+		retrier:      &notify.Retrier{},
+		webhookURL:   c.WebhookURL,
+		postJSONFunc: notify.PostJSON,
+	}
+
+	return n, nil
+}
+
+func (n *Notifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
+	key, err := notify.ExtractGroupKey(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	level.Debug(n.logger).Log("incident", key)
+
+	data := notify.GetTemplateData(ctx, n.tmpl, as, n.logger)
+	tmpl := notify.TmplText(n.tmpl, data, &err)
+	if err != nil {
+		return false, err
+	}
+
+	title := tmpl(n.conf.Title)
+	if err != nil {
+		return false, err
+	}
+	text := tmpl(n.conf.Text)
+	if err != nil {
+		return false, err
+	}
+
+	alerts := types.Alerts(as...)
+	color := colorGrey
+	switch alerts.Status() {
+	case model.AlertFiring:
+		color = colorRed
+	case model.AlertResolved:
+		color = colorGreen
+	}
+
+	var url string
+	if n.conf.WebhookURL != nil {
+		url = n.conf.WebhookURL.String()
+	} else {
+		content, err := os.ReadFile(n.conf.WebhookURLFile)
+		if err != nil {
+			return false, fmt.Errorf("read webhook_url_file: %w", err)
+		}
+		url = strings.TrimSpace(string(content))
+	}
+
+	// A message as referenced in https://learn.microsoft.com/en-us/connectors/teams/?tabs=text1%2Cdotnet#request-body-schema
+	t := teamsMessage{
+		Type: "message",
+		Attachments: []Attachment{
+			{
+				ContentType: "application/vnd.microsoft.card.adaptive",
+				ContentURL:  nil,
+				Content: Content{
+					Schema:  "http://adaptivecards.io/schemas/adaptive-card.json",
```

Comment:
The text in desktop view is cut off and does not take up the full width of the message.
On mobile, only the title appears.
https://learn.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/cards/cards-format?tabs=adaptive-md%2Cdesktop%2Cconnector-html#full-width-adaptive-card
<img width="659" alt="desktop_view" src="https://github.com/user-attachments/assets/6d28d568-2858-4acc-a52d-fbf14b451abf">
![mobile_view](https://github.com/user-attachments/assets/223aaa24-9754-4765-8918-d8e943cea5e5)